docs/release-info/migr-ansi-sql-null.md

```diff
@@ -0,0 +1,385 @@
+sidebar_label: SQL compliant mode
+---
+
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one
+  ~ or more contributor license agreements.  See the NOTICE file
+  ~ distributed with this work for additional information
+  ~ regarding copyright ownership.  The ASF licenses this file
+  ~ to you under the Apache License, Version 2.0 (the
+  ~ "License"); you may not use this file except in compliance
+  ~ with the License.  You may obtain a copy of the License at
+  ~
+  ~   http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+-->
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+In Apache Druid 28.0.0, the default [null handling](../querying/sql-data-types.md#null-values) mode changed to be compliant with the SQL standard.
+This guide provides strategies for Druid operators who rely on the legacy Druid null handling behavior in their applications to transition to SQL compliant mode.
+Legacy is scheduled to be removed from Druid. 
+
+## SQL compliant null handling in Druid
+
+As of Druid 28.0.0, Druid writes segments in a SQL compatible null handling mode by default.
+This means that Druid stores null values distinctly from empty strings for string dimensions and distinctly from 0 for numeric dimensions.
+
+This can impact your application behavior because SQL the standard defines any comparison to null to be unknown.
+According to this three-value logic, `x <> 'some value'` only returns non-null values.
+
+The default Druid configurations for SQL compatible null handling mode is as follows:
+
+* `druid.generic.useDefaultValueForNull=false`
+* `druid.expressions.useStrictBooleans=true`
+* `druid.generic.useThreeValueLogicForNativeFilters=true` 
+
+Follow the [Null handling tutorial](../tutorials/tutorial-sql-null.md) to learn how the default null handling works in Druid.
+
+## Legacy null handling and two-value logic
+
+Prior to Druid 28.0.0, Druid defaulted to a legacy mode which stored default values instead of nulls.
+In legacy mode, Druid segments created at ingestion time have the following characteristics:
+
+- String columns can not distinguish an empty string, '', from null.
+    Therefore, Druid treats them both as interchangeable values.
+- Numeric columns can not represent null valued rows.
+    Therefore Druid stores 0 instead of null. 
+
+The Druid configurations for the deprecated legacy mode are as follows:
+
+* `druid.generic.useDefaultValueForNull=true`
+* `druid.expressions.useStrictBooleans=false`
+* `druid.generic.useThreeValueLogicForNativeFilters=true`
+
+These configurations are deprecated and scheduled for removal.
```

Comment:
Do they have to actively remove these configs or if they leave them will they be ignored? IIRC, they'll be ignored, but we should be explicit on whether or not they need to actively do anything or not. e.g.

```
If these are in your configuration, they'll be ignored once the legacy mode is removed.
```

Reply:
i imagine we might make warning logs on startup if we detect the configs that no longer do anything